DragaliaAPI/Features/Item/ItemService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using DragaliaAPI.Features.Player;
+using DragaliaAPI.Models;
+using DragaliaAPI.Models.Generated;
+using DragaliaAPI.Services.Exceptions;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.MasterAsset;
+using DragaliaAPI.Shared.MasterAsset.Models.Trade;
+using Microsoft.EntityFrameworkCore;
+
+namespace DragaliaAPI.Features.Item;
+
+public class ItemService(
+    IItemRepository itemRepository,
+    ILogger<ItemService> logger,
+    IUserService userService
+) : IItemService
+{
+    public async Task<IEnumerable<ItemList>> GetItemList()
+    {
+        return (await itemRepository.Items.ToListAsync()).Select(
+            x => new ItemList(x.ItemId, x.Quantity)
+        );
+    }
+
+    public async Task<AtgenRecoverData> UseItems(IEnumerable<AtgenUseItemList> items)
+    {
+        logger.LogDebug("Processing items {@useItems}", items);
+
+        UseItemEffect effect = UseItemEffect.None;
+        int totalQuantity = 0;
+
+        foreach (AtgenUseItemList item in items)
+        {
+            UseItemData data = MasterAsset.UseItem[item.item_id];
+            if (effect == UseItemEffect.None)
+                effect = data.ItemEffect;
```

Comment:
are we expecting to only get items of the same effect?

Reply:
item/use_recovery_stamina doesn't return an array of AtgenRecoverData so the game prevents you from mixing them